main/fileUtil.js

```diff
@@ -135,6 +141,59 @@ function deleteFile(filePath) {
     });
 }
 
+
+/**
+ * Delete the folder at the given path.
+ *
+ * @param {string} folderPath the folder path to delete.
+ * @returns {Promise} promise that resolves if successful.
+ */
+function deleteDir(folderPath) {
+    return new Promise((resolve, reject) => {
+        rimraf(folderPath, error => {
+            if (error) {
+                reject(new Error(`Unable to delete ${folderPath}: ${error.message}`));
+            } else {
+                resolve();
+            }
+        });
+    });
+}
+
+/**
+ * Copy files from source to destination
+ *
+ * @param {string} src the path to source.
+ * @param {string} dest the path to destination.
+ * @returns {Promise} promise that resolves if successful.
+ */
+function copy(src, dest) {
+    return new Promise((resolve, reject) => {
+        ncp(src, dest, error => {
+            if (error) {
+                reject(new Error(`Unable to copy ${src}: ${error.message}`));
+            } else {
+                resolve();
+            }
+        });
+    });
+}
+
+/**
+ * Change file mode
+ *
+ * @param {string} src the path to source.
+ * @param {string} mode the mode to change.
+ * @returns {Promise} promise that resolves if successful.
+ */
+function chmodDir(src, mode) {
+    return new Promise(resolve => {
+        chmodr(src, mode, error => {
+            resolve(error);
```

Comment:
What I think here is that if the file exists it will change the mode and then resolve, however, if the file does not exist, it will still resolve. This works in my situation but seems weird in a normal logic, I will find a better way to solve it.